src/bayesnf/spatiotemporal.py

```diff
@@ -186,27 +187,103 @@ def get_input_scales(self) -> np.ndarray:
 
 
 class BayesianNeuralFieldEstimator:
-  """Base class for BNF estimator."""
+  """Base class for BayesNF estimators. This class should not be initialized
+  directly, but rather one of the three subclasses that implementing
+  different model learning procedures:
+
+  - [BayesianNeuralFieldVI](BayesianNeuralFieldVI.md), for
+    ensembles of surrogate posteriors from variational inference.
+
+  - [BayesianNeuralFieldMAP](BayesianNeuralFieldMAP.md), for
+    stochastic ensembles of maximum-a-posteriori estimates.
+
+  - [BayesianNeuralFieldMLE](BayesianNeuralFieldMLE.md), for
+    stochastic ensembles of maximum likelihood estimates.
+
+  All three classes share the same `__init__` method described below.
+  """
 
   _ensemble_dims: int
   _prior_weight: float = 1.0
   _scale_epochs_by_batch_size: bool = False
 
   def __init__(
       self,
+      *,
       feature_cols: Sequence[str],
       target_col: str,
-      timetype: str,
-      freq: str,
       seasonality_periods: Sequence[float | str] | None = None,
       num_seasonal_harmonics: Sequence[int] | None = None,
       fourier_degrees: Sequence[float] | None = None,
-      observation_model: str = 'NORMAL',
+      interactions: Sequence[tuple[int, int]] | None = None,
+      freq: str,
```

Comment:
needs a default, or to be listed earlier

Reply:
The signature starts with `self, *, ...`, which treats `freq` as a "required" kwarg.Does it matter if there is a `*`?

Reply:
wild. i knew about `*`, but not that you could interleave defaults and non-defaults. I think a user would still expect to see the "required" arguments first in a signature (but after the `*`), but up to you.